src/main/java/com/example/securitynew/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,47 @@
+package com.example.securitynew.controller;
+
+import com.example.securitynew.dto.cartitem.CartItemRequestDto;
+import com.example.securitynew.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.securitynew.service.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/cart")
+public class ShoppingCartController {
+    private ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/cart/{id}")
```

Comment:
Redundant `/cart`. You have @RequestMapping("/api/cart").

Reply:
not fixed